Libraries/LibWeb/Layout/Node.cpp

```diff
@@ -855,6 +856,22 @@ void NodeWithStyle::apply_style(const CSS::StyleProperties& computed_style)
 
     computed_values.set_fill_opacity(computed_style.fill_opacity());
 
+    if (auto const& stroke_dasharray_or_none = computed_style.property(CSS::PropertyID::StrokeDasharray); !stroke_dasharray_or_none.is_keyword()) {
+        auto const& stroke_dasharray = stroke_dasharray_or_none.as_stroke_dasharray();
+        Vector<Variant<CSS::LengthPercentage, CSS::NumberOrCalculated>> dashes;
+
+        for (auto const& value : stroke_dasharray.dashes()) {
+            if (value->is_length())
+                dashes.append(CSS::LengthPercentage { value->as_length().length() });
+            else if (value->is_percentage())
+                dashes.append(CSS::LengthPercentage { value->as_percentage().percentage() });
+            else if (value->is_number())
+                dashes.append(CSS::NumberOrCalculated { value->as_number().number() });
+        }
```

Comment:
You're missing a check for `value->is_math()`. I think the rest of your code already handles `calc()` correctly.

Reply:
I did try it, calc() roundtrips fine through this. I'll look into adding a test.

Reply:
Computed probably will roundtrip fine, but the used value will lose the calc()s here.

Reply:
That test isn't exercising this code. `apply_style()` is setting the `computed_values` but that's not what `getComputedStyle()` returns - it'll be internally calling `computed_style.property(CSS::PropertyID::StrokeDasharray)` again.

The result you're assigning with `set_stroke_dasharray()` below will be missing out these calc() values. That'll be visible once you use it for actual painting. But for now I think if you look in the inspector in the "Resolved Style" tab you'll see that too. (Not 100% sure, I don't remember how that is produced. This property might not show there at all.)

`is_length()` only returns true for a `LengthStyleValue`, and `calc()` is stored separately as a `CSSMathValue`.

Reply:
Before:

<img width="824" alt="Screenshot 2024-11-21 at 9 27 49 AM" src="https://github.com/user-attachments/assets/a21f6192-7fca-44e9-b0f5-62752825e604">

After:

<img width="824" alt="Screenshot 2024-11-21 at 9 27 36 AM" src="https://github.com/user-attachments/assets/04141d35-7902-454f-aca9-bfc22d23359b">

The latter matches chrome.

Done, thanks!